### ISSUE TITLE
Sparc 32 defaulting to returning undefined8 for void

### ISSUE THREAD

**Issue by mumbel** [NationalSecurityAgency/ghidra#6990]
**Describe the bug**
Upgraded an image.  Noticed uncommitted void return functions want to now return undefined8.

Using unknown logic for the low 32 or really high, but wants to use param_2 as the high 32-bits.  Inserting at least a param_1 and param_2 into the prototype even when it's zero arguments.

Previous ghidra versions were also broken though sometimes, converting void into undefined4, returning param_1 (and adding param_1 as an argument even when not present)

**Environment (please complete the following information):**
 - OS: win10
 - Java Version: 21.0.4
 - Ghidra Version: 11.2
 - Ghidra Origin: official GitHub distro

**Comment by mumbel** [comment]
No, nvmd.  It's def breaking lots of functions, not just void return.  It doesn't pick up the return 4-byte value and uses the previous o0 value

**Comment by emteere** [comment]
Are these assigned return and param types, or is the decompiler recovering the types on the fly?
I'm assuming you mean the decompiler is computing them on the fly and you haven't  committed the param/return from the decompiler.

64 or 32 bit sparc?

Do you have any example functions you could share?  Either bytes or copy/paste of listing?

**Comment by mumbel** [comment]
sorry @emteere  missed the reply.  pcodetest is showing it looks like in 11.2.1, built with sparc-suse-linux-gcc (14). i just did the basic analysis and `main` (and others) were showing `return CONCAT(param_1,param_2)` for undefined8

[SPARC_GCC_O0_pcodtest.txt](https://github.com/user-attachments/files/17655888/SPARC_GCC_O0_pcodtest.txt)
